tests/run/quote-tasty-position/quoted_1.scala

```diff
@@ -0,0 +1,8 @@
+import scala.quoted._
+
+import dotty.tools.dotc.quoted.Toolbox._
+
+object Lines {
+  inline def line(dummy: Int): Int = ~lineImpl('(dummy))
+  def lineImpl(dummy: Expr[Int]): Expr[Int] = (dummy.toTasty.pos.startLine + 1).toExpr
```

Comment:
Would that work?
```scala
object Lines {
  implicit object CurrentPos
  inline def line(implicit pos: CurrentPos.type): Int = ~lineImpl('(pos))
  def lineImpl(dummy: Expr[CurrentPos.type]): Expr[Int] = (dummy.toTasty.pos.startLine + 1).toExpr
}

import Lines._
object Test {
  def main(args: Array[String]): Unit = {
    println(line)
  }
}
```

Reply:
I tried that and it does not. The issue is that currently, the implicitly inferred tree does not have a position. We should add the position in the future to allow this use case to work.

Reply:
Even better, this one works and does not need an additional implicit

```scala
class LineNumber(val value: Int) {
  override def toString: String = value.toString
}

object LineNumber {
  implicit inline def line[T]: LineNumber = ~impl2('[T])

  def impl2[T](x: Type[T]): Expr[LineNumber] = {
    val pos = {
      val (tree, ctx) = x.toTasty
      tree.pos(ctx)
    }
    '(new LineNumber(~pos.startLine.toExpr))
  }
}
```

Note that now the context is needed. The important part is that the quoted type has the position we need.

```scala
def foo(implicit line: LineNumber): Unit = {
  println("foo " + line)
}
foo(line)
foo
```